### ISSUE TITLE
Port to Pyside6

### ISSUE THREAD

**Issue by fabianfreyer** [ehntoo/binaryninja-svd#9]
Binary ninja is now using pyside6:

```
Failed to import python plugin: community/ehntoo_binaryninjasvd: Qt 6 must import PySide6 instead of PySide2
Failed to load plugin: ehntoo_binaryninjasvd
```

**Comment by ehntoo** [comment]
Hm, apparently PySide6 doesn't yet include the QtWebEngineWidgets module. I'll have to put together a release without the SVD browser.

**Comment by fabianfreyer** [comment]
Yep. @psifertex - since I think you added that, is there something else that could be used instead?

**Comment by psifertex** [comment]
No and this is the one plugin that I'm genuinely sad about losing that capability for. There's no simple fixes unfortunately. 

Technically QT 6.2 does drop next week and we might move to it and it supposedly will have a cross-platform QWebEngine so it could go back into the product, but since it doubles the installer size so we're strongly considering not including it at all at that point.

Possible work arounds include:
 - (EASY): Just pop open a browser to the URL as well as the proper download folder so the user can easily save files to the appropriate locations
 - (HARDER) - Build a custom UI around scraping the URL and extracting the valid files using the existing download provider (https://gist.github.com/psifertex/6fbc7532f536775194edd26290892ef7#file-update_snippets-py-L15-L21)

**Comment by ehntoo** [comment]
Understood on not wanting to double the installer size - it seems a bit overkill to ship an entire web browser.

Doing some googling, I suppose I could wrap pyOCD's cmsis-pack-manager project - seems like it would add nontrivial value, and I could also automatically set up flash and RAM regions based on those described in the `pdsc`.

I'll try and find time this week to get a placeholder release out tomorrow that nukes the browser integration and updates structure API usage to match the new beta APIs, since it seems like the long-term solution for the integrated packfile support may be a little more effort.

**Comment by four0four** [comment]
As a stop-gap, [here's](https://gist.github.com/four0four/c4ba37b95241df3b75b40eba0d3b1e11) the rather messy set of changes sitting in my local copy. Just removes the browser entirely, and fixes up some API changes. Maybe it'll save someone time :)

**Comment by psifertex** [comment]
I just submitted a removal of the browser and updates for PySide 6 as #12.

**Comment by ehntoo** [comment]
Fixed in release v3.0.0. I'll open a new issue to track an improved svd manager using `cmsis-pack-manager`. Thanks for all the contributions, folks. :)
